### ISSUE TITLE
Sounds resume playing when exiting the mod menu with esc while paused

### ISSUE THREAD

**Issue by somefoolouthere** [Prospector/ModMenu#294]
While in a singleplayer world, if you pause the game, click the Mods button, and press the escape key to exit the mod menu, all in-game sounds (including music) will resume playing even though the game is still paused.

Mods list (just to be safe): 
![image](https://user-images.githubusercontent.com/19628205/126845139-d9cdc537-aea4-46ba-b920-ff29b371bb2e.png)

**Comment by jackassmc** [comment]
I can reproduce this on Minecraft 1.17.1 and Mod Menu 2.0.3

**Comment by jackassmc** [comment]
Thank you for your bug report! Fixed 1.16 and 1.17 versions will be available soon.

<hr>

The problem was  calling `super.onClose()` in ModsScreen.

`Screen.onClose()`:
```java
public void onClose() {
    this.client.openScreen((Screen)null);
}
```
`MinecraftClient.openScreen()`:
```java
public void openScreen(@Nullable Screen screen) {
// [...]
    if (screen == null)
        this.soundManager.resumeAll();
        this.mouse.lockCursor();
    }
// [...]
}
```
